eox_core/api/v1/serializers.py

```diff
@@ -46,3 +48,43 @@ class EdxappUserQuerySerializer(EdxappUserSerializer):
     """
 
     activate_user = serializers.BooleanField(default=False)  # We need to allow the api to activate users later on
+
+
+class EdxappEnrollmentAttributeSerializer(serializers.Serializer):
+    namespace = serializers.CharField()
+    name = serializers.CharField()
+    value = serializers.CharField()
+
+
+class EdxappCourseEnrollmentSerializer(serializers.Serializer):
+    """Serializes CourseEnrollment
+
+    Aggregates all data from the Course Enrollment table, and pulls in the serialization for
+    the Course Descriptor and course modes, to give a complete representation of course enrollment.
+
+    """
+
+    user = serializers.CharField(max_length=30, default="")
+    #created = serializers.DateTimeField(default=datetime.now)
+    is_active = serializers.BooleanField(default=True)
+    mode = serializers.CharField(max_length=100)
+
+    def validate(self, data):
+        """
+        Check that there are no issues with enrollment
+        """
+        errors = check_edxapp_enrollment_is_valid(**data)
+        if errors:
+            raise serializers.ValidationError(", ".join(errors))
+        return data
+
+
+class EdxappCourseEnrollmentQuerySerializer(EdxappCourseEnrollmentSerializer):
+    """
+    Handles the serialization of the context data required to create an enrollemnt
+    on different backends
+    """
+    username = serializers.CharField(max_length=30)
+    force_registration = serializers.BooleanField(default=False)
+    course_id = serializers.CharField(max_length=255)
+    enrollment_attributes = EdxappEnrollmentAttributeSerializer(many=True)
```

Comment:
@Ivanca do you have an example of what kind of enrollment_attributes this will hold?

Reply:
https://github.com/edx/edx-platform/blob/master/common/djangoapps/enrollment/views.py#L430-L435